### ISSUE TITLE
vue/no-unregistered-components not working for <script setup>

### ISSUE THREAD

**Issue by niaodan2b** [vuejs/eslint-plugin-vue#1674]
- [ √] I have tried restarting my IDE and the issue persists.
- [ √] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

- **ESLint version:** 6.7.2 
- **eslint-plugin-vue version:** 7.20.0
- **Node version:** 14.17.6
- **Operating System:** Windows 10

**Please show your full configuration:**
.eslintrc:
```json5
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: [
    'plugin:vue/vue3-essential',
    '@vue/standard',
    '@vue/typescript/recommended'
  ],
  parserOptions: {
    ecmaVersion: 2020
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'vue/no-unregistered-components': ['error', { ignorePatterns: ['router-view', 'router-link'] }],
    'vue/script-setup-uses-vars': 'error'
  }
}
```

**What did you do?**	
Home.vue:
```vue
<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <HelloWorld msg="Welcome to Your Vue.js + TypeScript App"/>
    <aaa />
  </div>
</template>

<script setup lang="ts">
import HelloWorld from '@/components/HelloWorld.vue' // @ is an alias to /src
</script>
```

**What did you expect to happen?**
Eslint reports aaa as unregistered component

**What actually happened?**
No lint errors found

**Repository to reproduce this issue**
https://github.com/niaodan2b/test-repo

**Comment by ota-meshi** [comment]
If you want a new rule, could you open an issue that suggests a rule to make it easier for someone to work on it? Or could you open a PR to add a rule?
